tests/test-extensions/metadata-extension/src/main/java/org/mule/test/metadata/extension/MetadataSourceWithCallbackParameters.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.test.metadata.extension;
+
+import static org.mule.test.metadata.extension.MetadataConnection.PERSON;
+
+import org.mule.runtime.api.connection.ConnectionProvider;
+import org.mule.runtime.api.exception.MuleException;
+import org.mule.runtime.extension.api.annotation.execution.OnError;
+import org.mule.runtime.extension.api.annotation.execution.OnSuccess;
+import org.mule.runtime.extension.api.annotation.metadata.MetadataKeyId;
+import org.mule.runtime.extension.api.annotation.metadata.MetadataScope;
+import org.mule.runtime.extension.api.annotation.metadata.TypeResolver;
+import org.mule.runtime.extension.api.annotation.param.Connection;
+import org.mule.runtime.extension.api.annotation.param.Parameter;
+import org.mule.runtime.extension.api.annotation.param.ParameterGroup;
+import org.mule.runtime.extension.api.runtime.source.Source;
+import org.mule.runtime.extension.api.runtime.source.SourceCallback;
+import org.mule.runtime.extension.api.runtime.source.SourceCallbackContext;
+import org.mule.runtime.extension.api.runtime.source.SourceCompletionCallback;
+import org.mule.tck.message.StringAttributes;
+import org.mule.test.metadata.extension.model.response.ErrorResponse;
+import org.mule.test.metadata.extension.model.response.SuccessResponse;
+import org.mule.test.metadata.extension.resolver.TestInputAndOutputWithAttributesResolverWithKeyResolver;
+import org.mule.test.metadata.extension.resolver.TestInputOutputSourceResolverWithKeyResolver;
+import org.mule.test.metadata.extension.resolver.TestInputResolver;
+import org.mule.test.metadata.extension.resolver.TestMetadataInputCarResolver;
+
+import java.util.Map;
+
+
+@MetadataScope(keysResolver = TestInputOutputSourceResolverWithKeyResolver.class,
+    outputResolver = TestInputAndOutputWithAttributesResolverWithKeyResolver.class)
+public class MetadataSourceWithCallbackParameters extends Source<Map<String, Object>, StringAttributes> {
+
+  public static boolean STARTED = false;
```

Comment:
![Vulnerability](https://sonarsource.github.io/sonar-github/vulnerability.png 'Vulnerability') Vulnerability: Make STARTED a static final constant or non-public and provide accessors if needed. ([squid:ClassVariableVisibilityCheck](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AClassVariableVisibilityCheck))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule&pullRequest=7668&issues=AWjdfEJHotIXnmsUGfbG&open=AWjdfEJHotIXnmsUGfbG)

[ISSUE_KEY]: AWjdfEJHotIXnmsUGfbG